4 (maint)/6 (master) test fails with super fresh (3 days old) git-annex

https://github.com/datalad/git-annex/runs/5292823639?check_suite_focus=true

```
======================================================================
FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 742, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/core/local/tests/test_save.py", line 648, in test_partial_unlocked
    assert_repo_status(ds.path)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 1624, in assert_repo_status
    % (state, state_files, oktobefound))
AssertionError: unexpected content of state "modified": [PosixPath('/tmp/datalad_temp_test_partial_unlockedgjlcmggh/culprit.txt')] != []

======================================================================
FAIL: datalad.core.local.tests.test_status.test_status_symlinked_dir_within_repo
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 742, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/core/local/tests/test_status.py", line 302, in test_status_symlinked_dir_within_repo
    call()
AssertionError: CommandError not raised

======================================================================
FAIL: datalad.local.tests.test_addurls.TestAddurls.test_addurls_metafail
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 742, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/local/tests/test_addurls.py", line 763, in test_addurls_metafail
    result_renderer=None)
AssertionError: IncompleteResultsError not raised

======================================================================
FAIL: datalad.support.tests.test_annexrepo.test_annexjson_protocol
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 563, in _wrap_with_tree
    return t(*(arg + (d,)), **kw)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/support/tests/test_annexrepo.py", line 2197, in test_annexjson_protocol
    protocol=AnnexJsonProtocol)
AssertionError: CommandError not raised

----------------------------------------------------------------------
Ran 1300 tests in [3767](https://github.com/datalad/git-annex/runs/5292823639?check_suite_focus=true#step:14:3767).414s

FAILED (SKIP=93, failures=4)
Error: Process completed with exit code 1.
```

<details>
<summary>and master got 6 failing</summary> 

```shell
======================================================================
35329
FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
35330
----------------------------------------------------------------------
35331
Traceback (most recent call last):
35332
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
35333
    self.test(*self.arg)
35334
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 874, in _wrap_with_tempfile
35335
    return t(*(arg + (filename,)), **kw)
35336
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/core/local/tests/test_save.py", line 650, in test_partial_unlocked
35337
    assert_repo_status(ds.path)
35338
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 1738, in assert_repo_status
35339
    % (state, state_files, oktobefound))
35340
AssertionError: unexpected content of state "modified": [PosixPath('/tmp/datalad_temp_test_partial_unlockedp49cit1q/culprit.txt')] != []
35341

35342
======================================================================
35343
FAIL: datalad.core.local.tests.test_status.test_status_symlinked_dir_within_repo
35344
----------------------------------------------------------------------
35345
Traceback (most recent call last):
35346
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
35347
    self.test(*self.arg)
35348
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 874, in _wrap_with_tempfile
35349
    return t(*(arg + (filename,)), **kw)
35350
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/core/local/tests/test_status.py", line 311, in test_status_symlinked_dir_within_repo
35401
----------------------------------------------------------------------
35402
Traceback (most recent call last):
35403
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
35404
    self.test(*self.arg)
35405
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 874, in _wrap_with_tempfile
35406
    return t(*(arg + (filename,)), **kw)
35407
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/local/tests/test_addurls.py", line 774, in test_addurls_metafail
35408
    result_renderer='disabled')
35409
AssertionError: IncompleteResultsError not raised
35410

35411
======================================================================
35412
FAIL: datalad.support.tests.test_annexrepo.test_annexjson_protocol
35413
----------------------------------------------------------------------
35414
Traceback (most recent call last):
35415
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
35416
    self.test(*self.arg)
35417
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 579, in _wrap_with_tree
35418
    return t(*(arg + (d,)), **kw)
35419
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/support/tests/test_annexrepo.py", line 2185, in test_annexjson_protocol
35420
    protocol=AnnexJsonProtocol)
35421
AssertionError: CommandError not raised
35422

35423
======================================================================
35424
FAIL: datalad.support.tests.test_fileinfo.test_get_file_annexinfo
35425
----------------------------------------------------------------------
35426
Traceback (most recent call last):
35427
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
35428
    self.test(*self.arg)
35429
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 579, in _wrap_with_tree
35430
    return t(*(arg + (d,)), **kw)
35431
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datalad/support/tests/test_fileinfo.py", line 346, in test_get_file_annexinfo
35432
    assert_raises(NoSuchPathError, repo.get_file_annexinfo, 'nothere')
35433
AssertionError: NoSuchPathError not raised by get_file_annexinfo
35434

35435
----------------------------------------------------------------------
35436
Ran 1373 tests in 3841.706s
35437

35438
FAILED (SKIP=79, failures=6)
```
</details>

Versions: annexremote=1.5.0 boto=2.49.0 cmd:7z=16.02 cmd:annex=10.20220222-1~ndall+1 cmd:bundled-git=2.30.2 cmd:git=2.30.2 cmd:system-git=2.35.1 cmd:system-ssh=8.2p1 humanize=4.0.0 iso8601=1.0.2 keyring=23.5.0 keyrings.alt=UNKNOWN msgpack=1.0.3 patoolib=1.12 platformdirs=2.5.1 requests=2.27.1 tqdm=4.62.3

so seems to correspond to a fresh release of git-annex unfortunately .  

<details>
<summary>seems started to appear 3 days ago</summary> 

```shell
(git)smaug:/mnt/datasets/datalad/ci/git-annex/builds/2022/02[master]
$> grep -Ra 'FAIL: datalad.core.local.tests.test_save.test_partial_unlocked'
cron-20220219/build-macos.yaml-595-85dd9355-failed/test-datalad (release)/11_Run datalad tests.txt:2022-02-19T04:07:29.9693860Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220219/build-macos.yaml-595-85dd9355-failed/test-datalad (master)/11_Run datalad tests.txt:2022-02-19T04:01:39.5815990Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220219/build-macos.yaml-595-85dd9355-failed/test-datalad (maint)/11_Run datalad tests.txt:2022-02-19T03:51:13.8345930Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220219/build-macos.yaml-595-85dd9355-failed/1_test-datalad (master).txt:2022-02-19T04:01:39.5816010Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220219/build-macos.yaml-595-85dd9355-failed/2_test-datalad (maint).txt:2022-02-19T03:51:13.8345940Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220219/build-macos.yaml-595-85dd9355-failed/3_test-datalad (release).txt:2022-02-19T04:07:29.9693870Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220219/build-ubuntu.yaml-583-85dd9355-failed/test-datalad (release)/14_Run datalad tests.txt:2022-02-19T03:56:12.2824740Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220219/build-ubuntu.yaml-583-85dd9355-failed/test-datalad (master)/14_Run datalad tests.txt:2022-02-19T04:01:21.7902184Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220219/build-ubuntu.yaml-583-85dd9355-failed/test-datalad (maint)/14_Run datalad tests.txt:2022-02-19T03:59:24.4669037Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220219/build-ubuntu.yaml-583-85dd9355-failed/1_test-datalad (master).txt:2022-02-19T04:01:21.7902194Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220219/build-ubuntu.yaml-583-85dd9355-failed/2_test-datalad (maint).txt:2022-02-19T03:59:24.4669041Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220219/build-ubuntu.yaml-583-85dd9355-failed/3_test-datalad (release).txt:2022-02-19T03:56:12.2824759Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-macos.yaml-596-85dd9355-failed/test-datalad (release)/11_Run datalad tests.txt:2022-02-20T03:59:46.1641580Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-macos.yaml-596-85dd9355-failed/test-datalad (master)/11_Run datalad tests.txt:2022-02-20T04:21:24.0459130Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-macos.yaml-596-85dd9355-failed/test-datalad (maint)/11_Run datalad tests.txt:2022-02-20T03:49:35.7382450Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-macos.yaml-596-85dd9355-failed/1_test-datalad (master).txt:2022-02-20T04:21:24.0459140Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-macos.yaml-596-85dd9355-failed/2_test-datalad (maint).txt:2022-02-20T03:49:35.7382460Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-macos.yaml-596-85dd9355-failed/3_test-datalad (release).txt:2022-02-20T03:59:46.1641580Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-ubuntu.yaml-584-85dd9355-failed/test-datalad (release)/14_Run datalad tests.txt:2022-02-20T03:59:42.4813134Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-ubuntu.yaml-584-85dd9355-failed/test-datalad (master)/14_Run datalad tests.txt:2022-02-20T04:07:06.8879676Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-ubuntu.yaml-584-85dd9355-failed/test-datalad (maint)/14_Run datalad tests.txt:2022-02-20T04:11:19.8988367Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-ubuntu.yaml-584-85dd9355-failed/1_test-datalad (master).txt:2022-02-20T04:07:06.8879684Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-ubuntu.yaml-584-85dd9355-failed/2_test-datalad (maint).txt:2022-02-20T04:11:19.8988406Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220220/build-ubuntu.yaml-584-85dd9355-failed/3_test-datalad (release).txt:2022-02-20T03:59:42.4813141Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220221/build-macos.yaml-597-85dd9355-failed/test-datalad (release)/11_Run datalad tests.txt:2022-02-21T04:23:04.9391210Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220221/build-macos.yaml-597-85dd9355-failed/test-datalad (master)/11_Run datalad tests.txt:2022-02-21T03:58:45.0953960Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220221/build-macos.yaml-597-85dd9355-failed/test-datalad (maint)/11_Run datalad tests.txt:2022-02-21T03:50:37.6780240Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220221/build-macos.yaml-597-85dd9355-failed/1_test-datalad (master).txt:2022-02-21T03:58:45.0953970Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220221/build-macos.yaml-597-85dd9355-failed/2_test-datalad (maint).txt:2022-02-21T03:50:37.6780250Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220221/build-macos.yaml-597-85dd9355-failed/3_test-datalad (release).txt:2022-02-21T04:23:04.9391220Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220221/build-ubuntu.yaml-585-85dd9355-failed/test-datalad (release)/14_Run datalad tests.txt:2022-02-21T03:54:10.3201919Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220221/build-ubuntu.yaml-585-85dd9355-failed/3_test-datalad (release).txt:2022-02-21T03:54:10.3201923Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-macos.yaml-598-85dd9355-failed/test-datalad (release)/11_Run datalad tests.txt:2022-02-22T04:16:58.1148250Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-macos.yaml-598-85dd9355-failed/test-datalad (master)/11_Run datalad tests.txt:2022-02-22T04:07:41.4377330Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-macos.yaml-598-85dd9355-failed/test-datalad (maint)/11_Run datalad tests.txt:2022-02-22T04:01:08.9866460Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-macos.yaml-598-85dd9355-failed/1_test-datalad (master).txt:2022-02-22T04:07:41.4377340Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-macos.yaml-598-85dd9355-failed/2_test-datalad (maint).txt:2022-02-22T04:01:08.9866470Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-macos.yaml-598-85dd9355-failed/3_test-datalad (release).txt:2022-02-22T04:16:58.1148260Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-ubuntu.yaml-586-85dd9355-failed/test-datalad (release)/14_Run datalad tests.txt:2022-02-22T03:57:09.4454167Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-ubuntu.yaml-586-85dd9355-failed/test-datalad (maint)/14_Run datalad tests.txt:2022-02-22T04:02:07.4484981Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-ubuntu.yaml-586-85dd9355-failed/test-datalad (master)/14_Run datalad tests.txt:2022-02-22T04:11:56.0010232Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-ubuntu.yaml-586-85dd9355-failed/1_test-datalad (master).txt:2022-02-22T04:11:56.0010237Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-ubuntu.yaml-586-85dd9355-failed/2_test-datalad (maint).txt:2022-02-22T04:02:07.4485258Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
cron-20220222/build-ubuntu.yaml-586-85dd9355-failed/3_test-datalad (release).txt:2022-02-22T03:57:09.4454171Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
manual-20220222/build-ubuntu.yaml-587-85dd9355-failed/test-datalad (release)/14_Run datalad tests.txt:2022-02-22T19:24:24.5820283Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
manual-20220222/build-ubuntu.yaml-587-85dd9355-failed/test-datalad (maint)/14_Run datalad tests.txt:2022-02-22T19:20:22.5890260Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
manual-20220222/build-ubuntu.yaml-587-85dd9355-failed/test-datalad (master)/14_Run datalad tests.txt:2022-02-22T19:21:09.4683815Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
manual-20220222/build-ubuntu.yaml-587-85dd9355-failed/1_test-datalad (master).txt:2022-02-22T19:21:09.4683819Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
manual-20220222/build-ubuntu.yaml-587-85dd9355-failed/2_test-datalad (maint).txt:2022-02-22T19:20:22.5890274Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked
manual-20220222/build-ubuntu.yaml-587-85dd9355-failed/3_test-datalad (release).txt:2022-02-22T19:24:24.5820287Z FAIL: datalad.core.local.tests.test_save.test_partial_unlocked

```
</details>

I guess it would have been nice to have git-annex releases to first ensure that no downstream testing in datalad is broken before a release (attn @joeyh)

--- a/datalad/core/local/tests/test_save.py
+++ b/datalad/core/local/tests/test_save.py
@@ -643,6 +643,8 @@ def test_partial_unlocked(path):
     ds.save()
     assert_repo_status(ds.path)
     # but now a change in the attributes
+    if '10.20220127' <= ds.repo.git_annex_version < '10.20220322':
+        raise SkipTest("annex bug https://git-annex.branchable.com/bugs/Change_to_annex.largefiles_leaves_repo_modified/")
     ds.unlock('culprit.txt')
     ds.repo.set_gitattributes([
         ('*', {'annex.largefiles': 'nothing'})])

--- a/datalad/core/local/tests/test_status.py
+++ b/datalad/core/local/tests/test_status.py
@@ -302,13 +302,20 @@ def test_status_symlinked_dir_within_repo(path):
     if ds.repo.git_annex_version < "8.20200522" or on_windows:
         # TODO: on windows even with a recent annex -- no CommandError is raised, TODO
         assert_result_count(call(), 0)
-    else:
+
+    elif ds.repo.git_annex_version < '10.20220127' or (
+            ds.repo.git_annex_version > '10.20220322' and not ds.config.getbool(
+                    section="annex", option="skipunknown", default=False)):
         # As of 2a8fdfc7d (Display a warning message when asked to operate on a
         # file inside a symlinked directory, 2020-05-11), git-annex will error.
-        #
-        # TODO: Consider providing better error handling in this case.
+        # However, there's an annex bug for a brief period, where it exits
+        # zero. Later on, it depends on how the reporting by annex looks like
+        # whether or not we fail or ignore that error. This in turn depends
+        # on the config queried in the condition.
         with assert_raises(CommandError):
             call()
+    else:
+        assert_result_count(call(), 0)
 
 
 @with_tempfile

--- a/datalad/core/local/tests/test_status.py
+++ b/datalad/core/local/tests/test_status.py
@@ -300,22 +300,27 @@ def test_status_symlinked_dir_within_repo(path):
                          on_failure="ignore", result_renderer='disabled')
 
     if ds.repo.git_annex_version < "8.20200522" or on_windows:
-        # TODO: on windows even with a recent annex -- no CommandError is raised, TODO
+        # TODO: on windows even with a recent annex -- no CommandError is
+        # raised, TODO
         assert_result_count(call(), 0)
-
-    elif ds.repo.git_annex_version < '10.20220127' or (
-            ds.repo.git_annex_version > '10.20220322' and not ds.config.getbool(
-                    section="annex", option="skipunknown", default=False)):
+    elif ds.repo.git_annex_version < '10.20220127':
         # As of 2a8fdfc7d (Display a warning message when asked to operate on a
         # file inside a symlinked directory, 2020-05-11), git-annex will error.
-        # However, there's an annex bug for a brief period, where it exits
-        # zero. Later on, it depends on how the reporting by annex looks like
-        # whether or not we fail or ignore that error. This in turn depends
-        # on the config queried in the condition.
         with assert_raises(CommandError):
             call()
-    else:
+    elif '10.20220127' <= ds.repo.git_annex_version < '10.20220322':
+        # No error on annex' side since 10.20220127;
+        # However, we'd now get something like this:
+        # > git annex find bar/f
+        # error: pathspec 'bar/f' did not match any file(s) known to git
+        # Did you forget to 'git add'?
+        #
+        # But exists zero until 10.20220322!
         assert_result_count(call(), 0)
+    else:
+        res = call()
+        assert_result_count(res, 1, status='error', state='unknown',
+                            path=str(bar_f))
 
 
 @with_tempfile

--- a/datalad/distributed/drop.py
+++ b/datalad/distributed/drop.py
@@ -793,7 +793,7 @@ def _postproc_annexdrop_result(res, respath_by_status, ds, **kwargs):
     respath_by_status[success] = \
         respath_by_status.get(success, []) + [res['path']]
     if res["status"] == "error" and res["action"] == "drop":
-        msg = res["message"]
+        msg = res.get("message", None)
         if isinstance(msg, str) and "Use --force to" in msg:
             # Avoid confusing datalad-drop callers with git-annex-drop's
             # suggestion to use --force.

--- a/datalad/distributed/tests/test_drop.py
+++ b/datalad/distributed/tests/test_drop.py
@@ -77,7 +77,7 @@ def test_drop_file_content(path, outside_path):
             type='file',
             status='error',
             action='drop',
-            message='not found',
+            error_message='File unknown to git',
             path=str(ds.pathobj / rp),
             refds=ds.path,
         )

--- a/datalad/support/annex_utils.py
+++ b/datalad/support/annex_utils.py
@@ -1,0 +1,67 @@
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Internal helper functions for interfacing git-annex
+"""
+
+from datalad.utils import ensure_list
+
+
+def _fake_json_for_non_existing(paths, cmd):
+    """Create faked JSON records for nonexisting paths provided by `paths`
+    after running `cmd`.
+
+    Internal helper for `AnnexRepo._call_annex_records`.
+
+    Parameters:
+    -----------
+    paths: str or list of str
+        paths to create annex-like JSON records for, communicating that the
+        path is unknown.
+    cmd: str
+        annex cmd for which to fake this result
+    """
+
+    return [{"command": cmd,
+             "file": f,
+             "note": "not found",
+             "success": False,
+             "error-messages": ["File unknown to git"]  # Note,
+             # that git's and annex' reporting here differs by config and
+             # command on whether they say "does not exist" or "did not match
+             # any file known to git".
+             } for f in ensure_list(paths)]
+
+
+def _get_non_existing_from_annex_output(output):
+    """This parses annex' output for messages about non-existing paths
+    and returns a list of such paths (as strings).
+
+    Internal helper for `_call_annex_records`.
+    """
+
+    # This should cease to exist whenever annex provides an actual API
+    # for retrieving such information;
+    # see https://git-annex.brachable.com/todo/api_for_telling_when_nonexistant_or_non_git_files_passed/
+
+    # Output largely depends on annex config annex.skipunknown
+    # (see https://github.com/datalad/datalad/pull/6510#issuecomment-1054499339)
+    # and git-annex' default of annex.skipunknown changed as of 10.20220127.
+    # However, that appears to not be true for all commands. annex-add would
+    # still report in the "git-annex: ... not found" fashion rather than
+    # "error:  ... did not match any file(s) known to git". Depends on
+    # what annex is calling internally *and* on that config. Apparently git
+    # itself isn't consistent in that regard.
+    # Therefore, account for both ways of reporting unknown paths.
+
+    unknown_paths = []
+    for line in output.splitlines():
+        if 'did not match any file(s) known to git' in line:
+            unknown_paths.append(line[17:-40])
+        elif line.startswith('git-annex:') and line.endswith(' not found'):
+            unknown_paths.append(line[11:-10])
+
+    return unknown_paths

--- a/datalad/support/annexrepo.py
+++ b/datalad/support/annexrepo.py
@@ -3311,12 +3311,23 @@ class AnnexRepo(GitRepo, RepoInterface):
             path = self.pathobj.joinpath(ut.PurePosixPath(j['file']))
             rec = info.get(path, None)
             if rec is None:
-                if init is not None:
+                # git didn't report on this path
+                if j.get('success', None) is False:
+                    # Annex reports error on that file. Create an error entry,
+                    # as we can't currently yield a prepared error result from
+                    # within here.
+                    rec = {'status': 'error', 'state': 'unknown'}
+                elif init is not None:
                     # init constraint knows nothing about this path -> skip
                     continue
-                rec = {}
+                else:
+                    rec = {}
             rec.update({'{}{}'.format(key_prefix, k): j[k]
-                       for k in j if k != 'file'})
+                       for k in j if k != 'file' and k != 'error_messages'})
+            # change annex' `error_messages` into singular to match result
+            # records:
+            if j.get('error-messages', None):
+                rec['error_message'] = '\n'.join(m.strip() for m in j['error-messages'])
             if 'bytesize' in rec:
                 # it makes sense to make this an int that one can calculate with
                 # with

--- a/datalad/support/annexrepo.py
+++ b/datalad/support/annexrepo.py
@@ -57,6 +57,10 @@ from datalad.runner.utils import (
 # must not be loads, because this one would log, and we need to log ourselves
 from datalad.support.json_py import json_loads
 from datalad.support.exceptions import CapturedException
+from datalad.support.annex_utils import (
+    _fake_json_for_non_existing,
+    _get_non_existing_from_annex_output,
+)
 from datalad.ui import ui
 import datalad.utils as ut
 from datalad.utils import (
@@ -1047,15 +1051,9 @@ class AnnexRepo(GitRepo, RepoInterface):
             )
         except CommandError as e:
             # Note: Workaround for not existing files as long as annex doesn't
-            # report it within JSON response:
+            # report it within JSON.
             # see http://git-annex.branchable.com/bugs/copy_does_not_reflect_some_failed_copies_in_--json_output/
-            not_existing = [
-                # cut the file path from the middle, no useful delimiter
-                # need to deal with spaces too!
-                line[11:-10] for line in e.stderr.splitlines()
-                if line.startswith('git-annex:') and
-                line.endswith(' not found')
-            ]
+            not_existing = _get_non_existing_from_annex_output(e.stderr)
             if not_existing:
                 if out is None:
                     # we create the error reporting herein. If all files were
@@ -1063,13 +1061,7 @@ class AnnexRepo(GitRepo, RepoInterface):
                     # anything
                     out = {'stdout_json': []}
                 out['stdout_json'].extend(
-                    {
-                        "command": args[0],
-                        "file": f,
-                        "note": "not found",
-                        "success": False,
-                    }
-                    for f in not_existing
+                    _fake_json_for_non_existing(not_existing, args[0])
                 )
 
             # Note: insert additional code here to analyse failure and possibly
@@ -1100,6 +1092,23 @@ class AnnexRepo(GitRepo, RepoInterface):
             #        "Running %s resulted in stderr output: %s",
             #        args, shorten(e.stderr)
             #    )
+
+        # git-annex fails to non-zero exit when reporting an error on
+        # non-existing paths in some versions and/or commands.
+        # Hence, check for it on non-failure, too. This became apparent with
+        # annex 10.20220127, but was a somewhat "hidden" issue for longer.
+        #
+        # Note, that this may become unnecessary after annex'
+        # ce91f10132805d11448896304821b0aa9c6d9845 (Feb 28, 2022)
+        # "fix annex.skipunknown false error propagation"
+        if 'stderr' in out:
+            not_existing = _get_non_existing_from_annex_output(out['stderr'])
+            if not_existing:
+                if out is None:
+                    out = {'stdout_json': []}
+                out['stdout_json'].extend(
+                    _fake_json_for_non_existing(not_existing, args[0])
+                )
 
         json_objects = out.pop('stdout_json')
 
@@ -2353,12 +2362,17 @@ class AnnexRepo(GitRepo, RepoInterface):
             else:
                 json_objects = _call_cmd(cmd, files)
 
+        # json_objects can contain entries w/o a "whereis" field. Unknown to
+        # git paths in particular are returned in such records. Code below is
+        # only concerned with actual whereis results.
+        whereis_json_objects = [o for o in json_objects if "whereis" in
+                                o.keys()]
 
         if output in {'descriptions', 'uuids'}:
             return [
                 [remote.get(output[:-1]) for remote in j.get('whereis')]
                 if j.get('success') else []
-                for j in json_objects
+                for j in whereis_json_objects
             ]
         elif output == 'full':
             # TODO: we might want to optimize storage since many remotes entries will be the
@@ -2371,7 +2385,7 @@ class AnnexRepo(GitRepo, RepoInterface):
                 else str(Path(PurePosixPath(j['file'])))
                 if on_windows else j['file']
                 : self._whereis_json_to_dict(j)
-                for j in json_objects
+                for j in whereis_json_objects
                 if not j.get('key', '').endswith('.this-is-a-test-key')
             }
 

--- a/datalad/support/tests/test_annexrepo.py
+++ b/datalad/support/tests/test_annexrepo.py
@@ -2165,6 +2165,7 @@ def test_error_reporting(path):
             # whole thing, despite space, properly quotes backslash
             'file': 'gl\\orious BS',
             'note': 'not found',
+            'error-messages': ['File unknown to git'],
             'success': False}]
     )
 

--- a/datalad/support/tests/test_annexrepo.py
+++ b/datalad/support/tests/test_annexrepo.py
@@ -1397,6 +1397,22 @@ def test_annex_copy_to(src, origin, clone):
             # That is not how annex behaves
             # http://git-annex.branchable.com/bugs/copy_does_not_reflect_some_failed_copies_in_--json_output/
             # for non-existing files output goes into stderr
+            #
+            # stderr output depends on config+version of annex, though:
+            if not dl_cfg.getbool(
+                    section="annex", option="skipunknown",
+                    # git-annex switched default for this config:
+                    default=bool(
+                        external_versions['cmd:annex'] < '10.20220127')):
+
+                stderr = "error: pathspec 'nonex1' did not match any file(s) " \
+                         "known to git\n" \
+                         "error: pathspec 'nonex2' did not match any file(s) " \
+                         "known to git\n"
+            else:
+                stderr = "git-annex: nonex1 not found\n" \
+                         "git-annex: nonex2 not found\n"
+
             raise CommandError(
                 "Failed to run ...",
                 stdout_json=[
@@ -1405,9 +1421,7 @@ def test_annex_copy_to(src, origin, clone):
                     {"command":"copy","note":"checking target ...",
                      "success":True, "key":"akey2", "file":"existed"},
                 ],
-                stderr=
-                    'git-annex: nonex1 not found\n'
-                    'git-annex: nonex2 not found\n'
+                stderr=stderr
             )
         else:
             return orig_run(command, **kwargs)
@@ -2178,15 +2192,26 @@ def test_annexjson_protocol(path):
     if logging.getLogger('datalad.annex').getEffectiveLevel() > 8:
         eq_(res['stderr'], '')
 
+    # Note: git-annex-find <non-existent-path> does not error with all annex
+    # versions. Fixed in annex commit
+    # ce91f10132805d11448896304821b0aa9c6d9845 (Feb 28, 2022).
+    if '10.20220127' < external_versions['cmd:annex'] < '10.20220322':
+        raise SkipTest("zero-exit annex-find bug")
+
     # now the same, but with a forced error
     with assert_raises(CommandError) as e:
-        res = ar._call_annex(
-            ['find', '.', 'error', '--json'],
-            protocol=AnnexJsonProtocol)
+        ar._call_annex(['find', '.', 'error', '--json'],
+                       protocol=AnnexJsonProtocol)
     # normal operation is not impaired
     eq_(e.exception.kwargs['stdout_json'], orig_j)
-    # we get a clue what went wrong
-    assert_in('error not found', e.exception.stderr)
+    # we get a clue what went wrong,
+    # but reporting depends on config + version (default changed):
+    msg = "pathspec 'error' did not match" if not dl_cfg.getbool(
+        section="annex", option="skipunknown",
+        # git-annex switched default for this config:
+        default=bool(external_versions['cmd:annex'] < '10.20220127')) else \
+        "error not found"
+    assert_in(msg, e.exception.stderr)
     # there should be no errors reported in an individual records
     # hence also no pointless statement in the str()
     assert_not_in('errors from JSON records', str(e.exception))

